### PR TITLE
Judge 2 method

### PR DESCRIPTION
**Pull request by wangmengbo**
1. extract pre-defined cell type used for coarse training as an argument.
1. complete viash pipeline testing in docker mode on my local machine.

### CODE REVIEW

File: scripts/add_a_method.sh [hunk 1]
```diff
@@ -8,7 +8,7 @@ exit 1
 scripts/download_resources.sh
 
 # create a new component
-method_id="my_method"
+method_id="scape"
```
Comment (rcannood) [inline scripts/add_a_method.sh:11]: Please undo these changes -- they should not be modified for everybody else to see :relaxed: 

```suggestion
method_id="my_method"
```

File: scripts/add_a_method.sh [hunk 2]
```diff
@@ -38,6 +38,14 @@ viash run src/task/methods/$method_id/config.vsh.yaml -- \
...
   --output "output/prediction.parquet"
 
+# run the method (using both parquest and h5ad as input)
+viash run src/task/methods/$method_id/config.vsh.yaml -- \
+  --de_train "resources/neurips-2023-kaggle/de_train.parquet" \
+  --de_train_h5ad "resources/neurips-2023-data/de_train.h5ad" \
+  --id_map "resources/neurips-2023-kaggle/id_map.csv" \
+  --output "output/prediction.parquet" \
+  --output_dir "output/neurips-2023-data/tmp_result.2"
+
```
Comment (rcannood) [inline scripts/add_a_method.sh:48]:
```suggestion
```

File: src/task/methods/scape/config.vsh.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      direction: output
+      description: Additional output directory for the model.
+      required: true
+      example: path/to/output_dir
+    - type: string
+      name: --cell
+      description: Pre-defined cell type held for pre-training.
+      required: false
+      example: NK cells
+      default: NK cells
```
Comment (rcannood) [inline src/task/methods/scape/config.vsh.yaml:23]: The unit test runs out of memory on the CI. Could you add one or more arguments to the component with a `default` and a `test_default`? The default will be what gets used during the actual benchmark, while `test_default` is used during CI testing. For example:

```suggestion
      default: NK cells
    - type: integer
      name: --epochs
      description: ...Add description here...
      default: 300
      info:
        test_default: 2
    - type: integer
      name: --epochs_enhanced
      description: ...Add description here...
      default: 300
      info:
        test_default: 2

File: src/task/methods/scape/script.py [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+## VIASH END
+print(f"par: {par}")
+
+if not os.path.isdir(par['output_dir']):
+	os.makedirs(par['output_dir'])
+
+# epochs = 2
+# epochs_enhanced = 2
+epochs = 300
+epochs_enhanced = 800
```
Comment (rcannood) [inline src/task/methods/scape/script.py:33]:
```suggestion
# fetch parameters
epochs = par["epochs"]
epochs_enhanced = par["epochs_enhanced"]

**Review by rcannood — CHANGES_REQUESTED** [review]
Thanks @wangmengbo ! I made a few suggestions. With these changes, hopefully the CI will pass!

**Comment by wangmengbo** [comment]
Hi, Robrecht. It seems the viash complains `test_default` being not correctly set. I'm not sure what's the exact reason causing viash cannot parse it. Any suggestion?

**Comment by rcannood** [comment]
> Hi, Robrecht. It seems the viash complains `test_default` being not correctly set. I'm not sure what's the exact reason causing viash cannot parse it. Any suggestion?

It's not Viash complaining, it was a mistake in the unit test I added for the method components. Should be fixed now!

**Comment by wangmengbo** [comment]
Thanks for fixing this. However, the test seems still be killed. I'll try do some gc during training.

**Review by rcannood — APPROVED** [review]

**Pull request merged**